modules/aws/eks/eks.tf

```diff
@@ -51,6 +54,63 @@ resource "aws_eks_cluster" "this" {
     }
   }
 
+  tags       = local.global_tags
+  depends_on = [aws_cloudwatch_log_group.this]
+}
+
+
+
+# CloudWatch Encrytion
+resource "aws_kms_key" "cloudwatch_encryption" {
+  description         = "Used for cloudwatch encryption"
+  enable_key_rotation = true
+
+  policy = <<EOF
+{
+  "Version" : "2012-10-17",
+  "Id" : "key-default-1",
+  "Statement" : [ {
+      "Sid" : "Enable IAM User Permissions",
+      "Effect" : "Allow",
+      "Principal" : {
+        "AWS" : "arn:aws:iam::${data.aws_caller_identity.current.account_id}:root"
+      },
+      "Action" : "kms:*",
+      "Resource" : "*"
+    },
+    {
+      "Effect": "Allow",
+      "Principal": { "Service": "logs.${data.aws_region.current.name}.amazonaws.com" },
+      "Action": [
+        "kms:Encrypt*",
+        "kms:Decrypt*",
+        "kms:ReEncrypt*",
+        "kms:GenerateDataKey*",
+        "kms:Describe*"
+      ],
+      "Resource": "*",
+      "Condition": {
+        "ArnEquals": {
+            "kms:EncryptionContext:aws:logs:arn": "arn:aws:logs:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:log-group:/aws/eks/${local.eks_cluster_name}/cluster"
```

Comment:
Surely the EKS resource knows its ARN? Maybe I'm reading it wrong, but it doesn't look like a reference here would be cyclical?

Reply:
Yes they do. The problem is that I need the KMS to be created first.
Then the creation of CloudWatch
And finally the creation of EKS.

So I need  to know the EKS arn before it exist, but since I can't do that I use this ugly workaround that I found.